provider/strategy.go

```diff
@@ -0,0 +1,23 @@
+package provider
+
+import (
+	"context"
+	"gx/ipfs/QmR8BauakNcBa3RbE4nbQu76PDiJgoQgz8AJdhJuiU4TAw/go-cid"
+	ipld "gx/ipfs/QmcKKBwfz6FyQdHR2jsXrrF6XeSBXYL86anmWNewpFpoF5/go-ipld-format"
+	"gx/ipfs/QmdURv6Sbob8TVW2tFFve9vcEWrSUgwPqeqnXyvYhLrkyd/go-merkledag"
+)
+
+func NewProvideAllStrategy(dag ipld.DAGService) Strategy {
+	return func(ctx context.Context, root cid.Cid) <-chan cid.Cid {
+		cids := make(chan cid.Cid)
+		go func() {
+			cids <- root
```

Comment:
Perhaps not super important but I would make this channel write into a select statement in case context is cancelled, so this go routine doesn't possibly get left around.

Reply:
Thanks! I went through and put all the necessary contexts in place.